### PR TITLE
[BUG] Changed the stored values for batchTC to dim-wise kl

### PR DESCRIPTION
**Pull request by linesd**
- Changed batchTC to store the dim-wise KL instead of the total correlation loss

### CODE REVIEW

File: disvae/models/losses.py [hunk 1]
```diff
@@ -423,10 +423,9 @@ def __call__(self, data, recon_batch, latent_dist, is_train, storer, latent_samp
             storer['tc_loss'].append(tc_loss.item())
             storer['dw_kl_loss'].append(dw_kl_loss.item())
 
-            # TODO Remove this when visualisation fixed
-            tc_loss_vec = (logqz - logqz_prodmarginals)
+            dw_kl_loss_vec = (logqz_prodmarginals - logpz)
             for i in range(latent_dist.size(1)):
-                storer['kl_loss_' + str(i)].append(tc_loss_vec[i].item())
+                storer['dw_kl_loss_' + str(i)].append(dw_kl_loss_vec[i].item())
```
Comment (linesd) [inline disvae/models/losses.py]: @alecokas  - please check that this name change doesn't affect visualisation. I seem to recall that you made it name agnostic.

Comment (alecokas) [reply disvae/models/losses.py]: `kl_loss_i` is dimension-wise, so I think keeping it with the original name is the right thing to do. That being said, the viz should still work

**Comment by YannDubs** [comment]
> Is the KL that you have called `dw_kl_loss_` different from the usual `kl_loss_` that we calculate for other models?

the name is miseading it is teh kl between the aggregate posterior instead of the usal posterior. I.e. KL[q(z)||p(z)] instead of KL[q(z|x)||p(z)]

**Comment by YannDubs** [comment]
so it is different. But we do compute all the terms needed to compute the usual KL, maybe I'll just save the usual KL instead to not compare apple and oranges

File: disvae/models/losses.py (hunk 1 above)
Comment (linesd) [reply disvae/models/losses.py]: Changed

**Comment by YannDubs** [comment]
please don't merge the PR, I'll take care of the rest (I had to change the loss file and will add those changes locally, then push all)

**Pull request closed without merging**